PCbuild/regen.vcxproj

```diff
@@ -0,0 +1,180 @@
+
+  </ImportGroup>
+  <Target Name="_RegenGrammar" BeforeTargets="Build">
+    <!-- Regenerate Include/graminit.h and Python/graminit.c from Grammar/Grammar using pgen-->
+    <Exec Command="&quot;$(OutDir)python$(PyDebugExt).exe&quot; -m Parser.pgen &quot;..\Grammar\Grammar&quot; &quot;..\Grammar\Tokens&quot; &quot;..\Include\graminit.h.new&quot; &quot;..\Python\graminit.c.new&quot;" />
+    <Copy SourceFiles="..\Include\graminit.h.new" DestinationFiles="..\Include\graminit.h">
+      <Output TaskParameter="CopiedFiles" ItemName="_UpdatedH" />
+    </Copy>
+    <Copy SourceFiles="..\Python\graminit.c.new" DestinationFiles="..\Python\graminit.c">
+      <Output TaskParameter="CopiedFiles" ItemName="_UpdatedC" />
+    </Copy>
+    <Warning Text="Grammar updated. You will need to rebuild pythoncore to see the changes." Condition="'@(_UpdatedH)' != '' and '@(_UpdatedC)' != '' and $(Configuration) == 'Debug'" />
+  </Target>
+  <Target Name="_RegenAST_H" AfterTargets="_RegenGrammar">
+    <!-- Regenerate Include/Python-ast.h using Parser/asdl_c.py -h -->
+    <Exec Command="&quot;$(OutDir)python$(PyDebugExt).exe&quot; &quot;..\Parser\asdl_c.py&quot; -h &quot;..\Include\Python-ast.h.new&quot; &quot;..\Parser\Python.asdl&quot;" />
+    <Copy SourceFiles="..\Include\Python-ast.h.new" DestinationFiles="..\Include\Python-ast.h">
+      <Output TaskParameter="CopiedFiles" ItemName="_UpdatedH" />
+    </Copy>
+    <Warning Text="Python-ast.h updated. You will need to rebuild pythoncore to see the changes." Condition="'@(_UpdatedH)' != '' and '@(_UpdatedC)' != '' and $(Configuration) == 'Debug'" />
+  </Target>
+  <Target Name="_RegenAST_C" AfterTargets="_RegenGrammar">
+    <!-- Regenerate Python/Python-ast.c using Parser/asdl_c.py -c -->
+    <Exec Command="&quot;$(OutDir)python$(PyDebugExt).exe&quot; &quot;..\Parser\asdl_c.py&quot; -c &quot;..\Python\Python-ast.c.new&quot; &quot;..\Parser\Python.asdl&quot;" />
+    <Copy SourceFiles="..\Python\Python-ast.c.new" DestinationFiles="..\Python\Python-ast.c">
+      <Output TaskParameter="CopiedFiles" ItemName="_UpdatedH" />
+    </Copy>
+    <Warning Text="Python-ast.c updated. You will need to rebuild pythoncore to see the changes." Condition="'@(_UpdatedH)' != '' and '@(_UpdatedC)' != '' and $(Configuration) == 'Debug'" />
+  </Target>
+  <Target Name="_RegenOpcodes" AfterTargets="_RegenAST_C">
+    <!-- Regenerate Include/opcode.h from Lib/opcode.py using Tools/scripts/generate_opcode_h.py-->
+    <Exec Command="&quot;$(OutDir)python$(PyDebugExt).exe&quot; ..\Tools\scripts\generate_opcode_h.py &quot;..\Lib\opcode.py&quot; &quot;..\Include\opcode.h.new&quot;" />
+    <Copy SourceFiles="..\Include\opcode.h.new" DestinationFiles="..\Include\opcode.h">
+      <Output TaskParameter="CopiedFiles" ItemName="_Updated" />
+    </Copy>
+    <Warning Text="Opcodes updated. You will need to rebuild pythoncore to see the changes." Condition="'@(_Updated)' != '' and $(Configuration) == 'Debug'" />
```

Comment:
The `Configuration=Debug` check is used because in release builds we raise an error (IIRC).

Reply:
Ah, I took out all error messages

Reply:
It's probably just best to error all the time, since the build really ought to be restarted if anything changes.